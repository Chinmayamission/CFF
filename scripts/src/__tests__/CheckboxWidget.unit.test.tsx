import React from "react";
import { shallow, mount, render } from 'enzyme';
import CustomForm from "src/form/CustomForm";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders checkbox with title and description', () => {
  let schema = {
    "type": "object",
    "title": "Form",
    "properties": {
      "acceptTerms": {
        "title": "Accept terms?",
        "description": "Description for accepting terms"
      }
    }
  };
  let uiSchema = {
    "interests": {
      "ui:widget": "checkboxes"
    }
  }
  const wrapper = render(
    <CustomForm schema={schema} uiSchema={uiSchema} />
  );
  // expect(wrapper.exists(".root_interests")).toEqual(true);
  expect(wrapper).toMatchSnapshot();
}); 