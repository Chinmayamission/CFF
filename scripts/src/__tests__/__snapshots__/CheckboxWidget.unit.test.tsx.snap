// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checkbox with title and description 1`] = `
<form
  class="rjsf"
>
  <div
    class="form-group field field-object"
  >
    <div
      class="container-fluid"
    >
      <h2
        class="ccmt-cff-form-title"
      >
        Form
      </h2>
      <div />
      <div
        class="row"
      >
        <div
          class="form-group field field-undefined  col-12"
        >
          <div>
            <label
              class="control-label"
              for="root_acceptTerms"
            >
              <span>
                Accept terms?
              </span>
            </label>
          </div>
          <div
            class="my-2"
            id="root_acceptTerms__description"
          >
            <div>
              Description for accepting terms
            </div>
          </div>
          <div
            class="unsupported-field"
          >
            <p>
              Unsupported field schema
              <span>
                 for field 
                <code>
                  root_acceptTerms
                </code>
              </span>
              <em>
                : Unknown field type undefined
              </em>
              .
            </p>
            <pre>
              {
  "title": "Accept terms?",
  "description": "Description for accepting terms"
}
            </pre>
          </div>
          <div />
          <div />
        </div>
      </div>
    </div>
    <div />
    <div />
  </div>
  <p>
    <button
      class="btn btn-info"
      type="submit"
    >
      Submit
    </button>
  </p>
</form>
`;
